invenio_communities/communities/records/systemfields/parent_community.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2022 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Community PID slug field."""
+from uuid import UUID
+
+from invenio_records.systemfields import SystemField
+
+from invenio_communities.proxies import current_communities
+
+
+def is_valid_uuid(value):
+    """Check if the provided value is a valid UUID."""
+    try:
+        UUID(str(value))
+        return True
+    except (ValueError, AttributeError, TypeError):
+        return False
+
+
+class ParentCommunityField(SystemField):
+    """System field for parent community."""
+
+    def __init__(self, key="parent"):
+        """Create a new ParentCommunityField instance."""
+        super().__init__(key=key)
+
+    def obj(self, instance):
+        """Get the access object."""
+        obj = self._get_cache(instance)
+        if obj is not None:
+            return obj
+
+        value = self.get_dictkey(instance)
+        if value:
+            obj = current_communities.service.record_cls.get_record(value["id"])
+        else:
+            obj = None
+
+        self._set_cache(instance, obj)
+        return obj
+
+    def set_obj(self, record, obj):
+        """Set the access object."""
+        if is_valid_uuid(obj):
+            try:
+                # Attempt to retrieve the community to confirm its existence
+                current_communities.service.record_cls.get_record(obj)
+                record["parent"] = dict(id=obj)
```

Comment:
```suggestion
                record["parent"] = dict(id=str(obj))
```

Question: Are we waiting some schema changes for this?

This looks like an issue because the type of obj would UUID, which would fail validation.

Reply:
Right, I guess we can accept both the string and the UUID, thanks for highlighting this!